doc/vlan/switchport-mode-support/Switchport Mode and VLAN CLI Enhancement.md

```diff
@@ -0,0 +1,760 @@
+
+**4.** Implementing Quality of Service (QoS): QoS allows users to prioritize traffic on the network based on the type of traffic and the needs of users. By configuring switch ports to operate in access mode in conjunction with QoS parameters available in SONiC, users can ensure that traffic from specific devices is prioritized over other traffic on the network.
+
+Overall, switch port mode access is a useful feature for controlling access to your network, improving network performance and segmenting a network into multiple broadcast domains.
+
+### Trunk Mode
+Here are some common use cases for switch port mode trunk:
+
+**1.** Connecting switches: When users need to connect two or more switches together, they can configure the switch ports to operate in trunk mode. This allows multiple VLANs to be transmitted over a single link, which can help to reduce the number of ports required and simplify network topology.
+
+**2.** Connecting to a router: Users can use switch port mode trunk to connect a switch to a router or other network device that supports VLANs. This allows the user to create multiple VLANs on the switch and route traffic between them using the router.
+
+**3.** Reducing congestion: By enabling trunking, you can reduce the number of ports required to connect switches together. This can help to reduce congestion and improve network performance by allowing more bandwidth to be dedicated to data traffic.
+
+**4.** Improving network scalability: Trunking also improves network scalability. By allowing multiple VLANs to be transmitted over a single port, you can increase the number of devices that can be connected to the network without needing additional switches.
+
+Overall, the switch port mode trunk is a useful feature for connecting switches and other network devices together, improving network performance, and providing redundancy. By enabling trunking, users can simplify network topology, reduce congestion, and increase network scalability.
+
+## Architecture Design
+
+The overall SONiC architecture will remain the same and no new sub-modules will be introduced. Changes are made only in the CLI container and Config_DB.
+
+![Architecture](https://user-images.githubusercontent.com/61490193/217232727-811b98a8-ce9c-4227-a94a-1c83a4825a62.png)
+
+
+## High-level Design
+
+In this section we will explain sequence diagrams for the implemented features.
+
+### State Transition Diagram of Switchport modes for PORT or PORT CHANNEL
+
+The following state transition depicts the behavior of Switchport  modes access and trunk for adding port modes on a Port or PortChannel. Switching between these modes is also depicted here. Default port mode is “routed”.
+
+![State-Transition-Diagram](https://user-images.githubusercontent.com/61490193/223195784-05cc4e98-8883-4911-bca3-fded6a87d202.png)
```

Comment:
This diagram conflicts with what you have states under this section:

Examples/Usage of Switchport Modes Command
Following example shows usage of switchport modes “access” and “trunk” and switching of port mode from routed to access/trunk. Before switching mode from routed to access/trunk, the user must remove IP first. After the mode has been switched to access/trunk there will be no untagged/tagged vlan member(s) assigned

So when you move from trunk to access or vice versa, do youy keep the "untagged" membership or that is also removed?  the diagram seems to say it retains the "untagged" membership.  I actually like retaining the untagged membership and only force the user to have to remove the  tagged memebrship when moving from trunk to access...

Reply:
Yes, untagged member will be retained when moving from access to trunk or vice versa. Modified this section, as per your concern.